common/src/main/java/com/wynntils/core/text/StyledText.java

```diff
@@ -0,0 +1,487 @@
+        List<ClickEvent> clickEvents = new LinkedList<>();
+        List<HoverEvent> hoverEvents = new LinkedList<>();
+
+        final int length = texts.length;
+        for (int i = 0; i < length; i++) {
+            StyledText text = texts[i];
+            parts.addAll(text.parts);
+
+            if (i != length - 1) {
+                // Possible micro-optimization: If the separator does not have formatting, use the normal constructor
+                parts.add(StyledTextPart.fromComponentWithPossibleFormattingCodes(separator, Style.EMPTY, null, null));
+            }
+
+            clickEvents.addAll(text.clickEvents);
+            hoverEvents.addAll(text.hoverEvents);
+        }
+
+        return new StyledText(parts, clickEvents, hoverEvents);
+    }
+
+    public static StyledText join(String separator, Iterable<StyledText> texts) {
+        return join(separator, Iterables.toArray(texts, StyledText.class));
+    }
+
+    public static StyledText concat(StyledText... texts) {
+        List<StyledTextPart> parts = new LinkedList<>();
+        List<ClickEvent> clickEvents = new LinkedList<>();
+        List<HoverEvent> hoverEvents = new LinkedList<>();
+
+        for (StyledText text : texts) {
+            parts.addAll(text.parts);
+            clickEvents.addAll(text.clickEvents);
+            hoverEvents.addAll(text.hoverEvents);
+        }
+
+        return new StyledText(parts, clickEvents, hoverEvents);
+    }
+
+    public static StyledText concat(Iterable<StyledText> texts) {
+        return concat(Iterables.toArray(texts, StyledText.class));
+    }
+
+    public StyledText getNormalized() {
+        return new StyledText(
+                parts.stream().map(StyledTextPart::asNormalized).collect(Collectors.toList()),
+                temporaryWorkaround,
+                clickEvents,
+                hoverEvents);
+    }
+
+    public StyledText trim() {
+        return new StyledText(
+                parts.stream().map(StyledTextPart::trim).collect(Collectors.toList()),
+                temporaryWorkaround,
+                clickEvents,
+                hoverEvents);
+    }
+
+    public boolean isEmpty() {
+        return parts.stream().allMatch(StyledTextPart::isEmpty);
+    }
+
+    public boolean isBlank() {
+        return parts.stream().allMatch(StyledTextPart::isBlank);
+    }
+
+    public boolean contains(String string) {
+        return contains(string, PartStyle.StyleType.DEFAULT);
+    }
+
+    public boolean contains(StyledText styledText) {
+        return contains(styledText.getString(PartStyle.StyleType.DEFAULT), PartStyle.StyleType.DEFAULT);
+    }
+
+    public boolean contains(String string, PartStyle.StyleType styleType) {
+        return getString(styleType).contains(string);
+    }
+
+    public boolean contains(StyledText styledText, PartStyle.StyleType styleType) {
+        return contains(styledText.getString(styleType), styleType);
+    }
+
+    public boolean startsWith(String string) {
```

Comment:
Can you rename all these `String string` to  `String codedString` to clarify that this is what is expected here?